### PR TITLE
[SPARK-6591][SQL]Python data source load options should auto convert common types into strings

### PR DESCRIPTION
**Pull request by yjshen**
JIRA: https://issues.apache.org/jira/browse/SPARK-6591

### CODE REVIEW

**Comment by rxin** [comment]
cc @davies

File: python/pyspark/sql/readwriter.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def schema(self, schema):
     def option(self, key, value):
         """Adds an input option for the underlying data source.
         """
-        self._jreader = self._jreader.option(key, value)
+        self._jreader = self._jreader.option(key, str(value))
```
Comment (davies) [inline python/pyspark/sql/readwriter.py]: True will become `'True'`, but JVM may expect `'true'`

File: python/pyspark/sql/readwriter.py [hunk 2]
```diff
@@ -97,7 +97,8 @@ def load(self, path=None, format=None, schema=None, **options):
         :param schema: optional :class:`StructType` for the input schema.
         :param options: all other string options
 
-        >>> df = sqlContext.read.load('python/test_support/sql/parquet_partitioned')
+        >>> df = sqlContext.read.load('python/test_support/sql/parquet_partitioned', opt1 = True,
+        ...     opt2 = 1, opt3 = 'str')
```
Comment (davies) [inline python/pyspark/sql/readwriter.py]: `opt1=True`

Comment (rxin) [inline python/pyspark/sql/readwriter.py]: to be more explicit, the style guide in Python is to not put spaces around = for named parameters.

File: python/pyspark/sql/readwriter.py [hunk 3]
```diff
@@ -77,7 +77,7 @@ def schema(self, schema):
     def option(self, key, value):
         """Adds an input option for the underlying data source.
         """
-        self._jreader = self._jreader.option(key, value)
+        self._jreader = self._jreader.option(key, str(value).lower())
```
Comment (davies) [inline python/pyspark/sql/readwriter.py]: We can't just call `lower()` for str, only need that for boolean,

Comment (yjshen) [inline python/pyspark/sql/readwriter.py]: Can I do this way? Above the definition of `class DataFrameReader(object):`
```python
def to_str(value):
    """
    A wrapper over str(), but convert bool values to lower case string
    """
    if isinstance(value, bool):
        return str(value).lower()
    else:
        return str(value)
```

**Comment by davies** [comment]
LGTM, merging this into master and 1.5 branch, thanks!

**Pull request closed without merging**